extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerBuildSupport.java

```diff
@@ -152,34 +154,43 @@ public void updateClasspath(IProject project, IProgressMonitor monitor) throws C
                 ResourcesResult resourcesResult = buildServer.buildTargetResources(
                     new ResourcesParams(Arrays.asList(buildTarget.getId()))).join();
                 List<IClasspathEntry> resourceEntries = getResourceEntries(project, resourcesResult, resourceOutputFullPath, isTest);
-                classpath.addAll(resourceEntries);
+                for (IClasspathEntry entry : resourceEntries) {
+                    classpathMap.putIfAbsent(entry.getPath(), entry);
+                }
             }
         }
 
         // Source set can have its own Java version setting, we need to find the highest version.
         // Since in JDT, each project can only has one JDK.
         JvmBuildTargetEx jvmBuildTarget = getJvmTargetWithHighestSourceCompatibility(buildTargets);
         String highestJavaVersion = getHighestCompatibleJavaVersion(jvmBuildTarget.getGradleVersion());
-        IVMInstall vm = EclipseVmUtil.findOrRegisterStandardVM(jvmBuildTarget.getJavaVersion(),
-                highestJavaVersion, new File(jvmBuildTarget.getJavaHome()));
-        classpath.add(JavaCore.newContainerEntry(JavaRuntime.newJREContainerPath(vm)));
+        try {
+            IVMInstall vm = EclipseVmUtil.findOrRegisterStandardVM(jvmBuildTarget.getJavaVersion(),
+                    highestJavaVersion, new File(new URI(jvmBuildTarget.getJavaHome())));
```

Comment:
This change is to make sure the `javaHome` string is correctly parsed.

The `javaHome` string is a uri string, previously, I wrongly used it as a file path